server/api/documents.js

```diff
@@ -407,11 +407,15 @@ async function loadDocument({ id, shareId, user }) {
       authorize(user, "read", document);
     }
   } else {
-    document = await Document.findByPk(
-      id,
-      user ? { userId: user.id } : undefined
-    );
-    authorize(user, "read", document);
+    document = await Document.findByPk(id, {
+      userId: user ? user.id : undefined,
+      paranoid: false,
+    });
+    if (document.deletedAt) {
+      authorize(user, "restore", document);
+    } else {
+      authorize(user, "read", document);
+    }
```

Comment:
🍖 allow individual docs to be loaded if they are deleted and the user can restore them

Reply:
cool cool, was thinking of a similar solution